identity-apps-core/apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -129,57 +174,151 @@
+            RecoveryApiV2 recoveryApiV2 = new RecoveryApiV2();
+            String recoveryCode = request.getAttribute("recoveryCode") != null 
+                                ? (String) request.getAttribute("recoveryCode") 
+                                : Encode.forJava(request.getParameter("recoveryCode")); 
+
+            String usernameRecoveryOption = request.getAttribute("usernameRecoveryOption") != null 
+                        ? (String) request.getAttribute("usernameRecoveryOption") 
+                        : Encode.forJava(request.getParameter("usernameRecoveryOption"));
+
+            String isUserFoundParam = request.getParameter("isUserFound");
+            Boolean isUserFound = (isUserFoundParam != null) 
+                                ? Boolean.parseBoolean(isUserFoundParam) 
+                                : IdentityManagementEndpointUtil.getBooleanValue(request.getAttribute("isUserFound"));
+
+            String recoveryChannelType = null;
+            String recoveryChannelId = null;
+            
+            // Extract the recovery channel name and id from the recovery option.
+            if (usernameRecoveryOption != null) {
+                String[] parts = usernameRecoveryOption.split(":");
+                recoveryChannelId = parts[0];
+                recoveryChannelType = parts[1];
+            }
+            
+            request.setAttribute("recoveryChannelType", recoveryChannelType);
+ 
+            // Sending the notification if we only found a user.
+            if (isUserFound){
+                RecoveryRequest recoveryRequest = new RecoveryRequest();
+                recoveryRequest.setRecoveryCode(recoveryCode);
+                recoveryRequest.setChannelId(recoveryChannelId);
+
+                Map<String, String> requestHeaders = new HashedMap();
+                if (request.getParameter("g-recaptcha-response") != null) {
+                    requestHeaders.put("g-recaptcha-response", Encode.forJava(request.getParameter("g-recaptcha-response")));
+                }
 
-            IdentityManagementEndpointUtil.addErrorInformation(request, e);
-            request.getRequestDispatcher("recoveraccountrouter.do").forward(request, response);
+                try{
+                    recoveryApiV2.recoverUsername(recoveryRequest, tenantDomain, requestHeaders);
+                    
+                } catch (ApiException e) {
+                    IdentityManagementEndpointUtil.addErrorInformation(request, e);
+                    request.getRequestDispatcher("recoveraccountrouter.do").forward(request, response);
+                    return;
+                }
+            }
+            
+            request.getRequestDispatcher("username-recovery-complete.jsp").forward(request, response);
             return;
+
+        } else {
+            request.setAttribute("errorMsg", IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
```

Comment:
need to update the i18n usage function.

Reply:
Fixed by https://github.com/wso2/identity-apps/pull/7005/commits/150c0a6b01d6cdf2ad7aa111ecdc18242e2e7b0f